### PR TITLE
move auth checking code in commons, so we can reuse it in other projects

### PR DESCRIPTION
**Pull request by troggy**
Please remember the [Contributing Guidelines](https://WhiteRabbitTheWorldIsYours/dashboard/blob/master/CONTRIBUTION.md) :heart:

## What does this PR do?

- creates `verifyAuthToken` in dashboard-common package. Using it in BE
- fixes issue with dashboard-common build (follow up for apparently incomplete #63)

### CODE REVIEW

File: common/src/auth.ts [hunk 1]
```diff
@@ -46,4 +49,26 @@ export const createAuthMessage = ({ timestamp }: AuthMessageArgs) => ({
   message: {
     timestamp,
   },
-});
\ No newline at end of file
+});
+
+export const recoverSigner = ({ timestamp }: AuthMessageArgs, signature: string): string => {
```
Comment (johannbarbie) [inline common/src/auth.ts:54]: would be awesome to have some unit-tests here

Comment (troggy) [reply common/src/auth.ts:54]: agree. For now, this code is checked in unit tests in payment-backend :)

Expiration: https://github.com/WhiteRabbitTheWorldIsYours/payment-backend/pull/10/files#diff-fec52b85b347d9158977f0ff45dcb433R90

Recover signer: https://github.com/WhiteRabbitTheWorldIsYours/payment-backend/pull/10/files#diff-fec52b85b347d9158977f0ff45dcb433R108

Comment (troggy) [reply common/src/auth.ts:54]: adding some tests real quick

Comment (troggy) [reply common/src/auth.ts:54]: s'il vous plaît

Comment (johannbarbie) [reply common/src/auth.ts:56]: merci :)

**Pull request merged**